workflows/nanopore.nf

```diff
@@ -0,0 +1,214 @@
+def modules = params.modules.clone()
+
+if (params.input) { ch_input= file(params.input) }
+
+
+include { MULTIQC_TSV_FROM_LIST as MULTIQC_TSV_NO_SAMPLE_NAME     } from '../modules/local/multiqc_tsv_from_list'
+include { MULTIQC_TSV_FROM_LIST as MULTIQC_TSV_NO_BARCODES        } from '../modules/local/multiqc_tsv_from_list'
+include { MULTIQC_TSV_FROM_LIST as MULTIQC_TSV_BARCODE_COUNT      } from '../modules/local/multiqc_tsv_from_list'
+include { MULTIQC_TSV_FROM_LIST as MULTIQC_TSV_GUPPYPLEX_COUNT    } from '../modules/local/multiqc_tsv_from_list'
+include { PREPARE_NCBI_ACCESSION_ID                               } from '../modules/local/prepare_ncbi_accession_id'
+include { IRMA                                                    } from '../modules/local/irma'
+include { SUBTYPING_REPORT                                        } from '../modules/local/subtyping_report'
+include { COVERAGE_PLOT                                           } from '../modules/local/coverage_plot'
+include { VCF_FILTER_FRAMESHIFT                                   } from '../modules/local/vcf_filter_frameshift'
+include { MEDAKA                                                  } from '../modules/local/medaka'
+include { MAP                                                     } from '../modules/local/map'
+include { BCF_FILTER; BCF_CONSENSUS                               } from '../modules/local/bcftools'
+include { MOSDEPTH_GENOME                                         } from '../modules/local/mosdepth'
+include { CAT_FASTQ                                               } from '../modules/local/misc'
+include { BLAST_BLASTDBCMD                                        } from '../modules/local/pull_references'
```

Comment:
It seems that with seqtk we have to provide full sequence name, it does not work if provide accession/ref id. So I think we should we blastdbcmd insetad

Reply:
I forgot how unusual the `influenza.fna.gz` FASTA entry names were:

```
>gi|58576|gb|X52226|Influenza A virus (A/FPV/Rostock/34(H7N1)) gene for neuraminidase, genomic RNA
```

instead of like regular GenBank/RefSeq names like: 
```
>X52226 Influenza A virus (A/FPV/Rostock/34(H7N1)) gene for neuraminidase, genomic RNA
```

Creating a separate BLAST DB just for parsing a few sequences is high overhead. 

For seqtk, you'd have to provide "gi|58576|gb|X52226|Influenza" to retrieve the sequence since that's what's being parsed as the seq id.

How about when decompressing the `influenza.fna.gz` file, the fasta headers are "fixed"?

```bash
zcat influenza.fna.gz | sed -E 's/^>gi\|[0-9]+\|gb\|(\w+)\|(.*)/>\1 \2/' > influenza.fna
```

Then you should be able to use seqtk to pull the seq from the fasta easily:

```bash
echo "$seqid" | seqtk influenza.fna - > $ref_fasta
```

There might need to be some changes in other parts of the codebase for this change, e.g. removing this regex:

https://github.com/nhhaidee/nf-iav-illumina/blob/b21b3df5ad45ea1f3f140baa93caa137ef7ffc93/bin/parse_influenza_blast_results.py#L207

Reply:
Used Seqtk for workflow